### PR TITLE
Add minimum number of cells (3) for merge

### PR DESCRIPTION
**Pull request by jashapiro**
Following #799, here I am also filtering the merged data to not include samples with 2 or fewer cells. This turns out to be necessary for the calculation of tehe grouped PCA (at least that is what I think is going on).

I took a bit of a belt and suspenders approach here, checking both in the metadata during the workflow itself and in the R script where I check the SCEs before merging.

There is also a little cleanup in one of the `Util` functions so it doesn't throw an error if a file doesn't exist.

I did a small test with this and it seemed to work, but a larger test is still in progress, which is why I am filing this as a draft.

### CODE REVIEW

**Comment by jashapiro** [comment]
Tests passed, ready for review!

File: bin/merge_sces.R [hunk 1]
```diff
@@ -132,6 +132,18 @@ read_trim_sce <- function(sce_file) {
...
 
+# filter out libraries with fewer than 3 cells (causes errors with PCA)
+n_cells <- sce_list |> purrr::map_int(ncol)
+included_libs <- names(sce_list)[which(n_cells >= 3)]
+if (length(included_libs) < length(sce_list)) {
+  message(
+    "The following libraries have fewer than 3 cells and will be excluded from the merged object: ",
+    paste(setdiff(names(sce_list), included_libs), collapse = ", ")
+  )
+}
```
Comment (sjspielman) [inline bin/merge_sces.R:143]: might do this comparison first and use it for check? up to you

```suggestion
lib_diff <- setdiff(names(sce_list), included_libs)
if (length(lib_diff) > 0) {
  message(
    "The following libraries have fewer than 3 cells and will be excluded from the merged object: ",
    paste(lib_diff, collapse = ", ")
  )
}
```

File: merge.nf [hunk 2]
```diff
@@ -199,22 +199,32 @@ workflow {
...
         it.library_id,
-        file("${params.results_dir}/${it.project_id}/${it.sample_id}/${it.library_id}_processed.rds")
+        file("${params.results_dir}/${it.project_id}/${it.sample_id}/${it.library_id}_processed.rds"),
+        file("${params.results_dir}/${it.project_id}/${it.sample_id}/${it.library_id}_metadata.json")
       ]}
-      // only include libraries that have been processed through scpca-nf and aren't empty
-      .filter{it[2].exists() && it[2].size() > 0}
+      // only include libraries that have been processed through scpca-nf and have at least 3 cells
+      .filter{it[2].exists() && it[2].size() > 0 && Utils.getMetaVal(it[3], "processed_cells") >= 3}
+      .map{it[0..2]} // remove metadata file from tuple
```
Comment (sjspielman) [inline merge.nf:227]: I gather `0..2` syntax is the groovy equivalent of R's `0:2` (...ok, `1:3` what with indexing from 1 😄 )

**Review by sjspielman — APPROVED** [review]
LGTM, just one suggestion for R code if you want it!

File: merge.nf (hunk 2 above)
Comment (jashapiro) [reply merge.nf:227]: That is correct. I could have done a number of things to get rid of the last element. Probably the best is `it.dropRight(1)` to remove the last element, but this seemed fine here.

**Pull request merged**